### ISSUE TITLE
Snapshot properties (fetching data) not working after query.

### ISSUE THREAD

**Issue by gNiKnet** [firebase/firebase-admin-node#66]
### [READ] Step 1: Are you in the right place?

  * For issues or feature requests related to __the code in this repository__
    file a Github issue.
    * If this is a __feature request__ make sure the issue title starts with "FR:".
  * For general technical questions, post a question on [StackOverflow](http://stackoverflow.com/)
    with the firebase tag.
  * For general Firebase discussion, use the [firebase-talk](https://groups.google.com/forum/#!forum/firebase-talk)
    google group.
  * For help troubleshooting your application that does not fall under one
    of the above categories, reach out to the personalized
    [Firebase support channel](https://firebase.google.com/support/).

### [REQUIRED] Step 2: Describe your environment

   "windows": "10"
   "node.js": "8.2.1"
  "firebase-admin": "~4.2.1"

### [REQUIRED] Step 3: Describe the problem

I already posted this on StackOverflow, but I think that this is not really a question, rather an issue because it works in the documentation, but not in my project. https://stackoverflow.com/questions/45543072/firebase-admin-database-snapshot-data-after-query

#### Steps to reproduce:

I am trying to use a query to select a child and then retrieve the data from there, but I am unable to access the properties, i.e. `snapshot.val().property`.

#### Relevant Code:

JSON (data)
```
values: {
    a: {
        randomValue: "set",
        sorter: 1
    },
    b: {
        randomValue: "hat",
        sorter: 2
    }
}
```

```
admin.database().ref('values/a').once('value').then(snapshot => {
    console.log(snapshot.val().randomValue + ' .. ' + snapshot.val().sorter);
});
```
output:
set .. 1

```
admin.database().ref('values').orderByChild('sorter').equalTo(1).once('value').then(snapshot => {
    console.log(snapshot.val());
    console.log(snapshot.val().randomValue + ' .. ' + snapshot.val().sorter);
    console.log(snapshot.child('randomValue').val() + ' .. ' + snapshot.child('sorter').val());
}
```
output:
a: {
    randomValue: "set",
    sorter: 1
}
undefined .. undefined
null .. null

should be:
a: {
    randomValue: "set",
    sorter: 1
}
set .. 1
set .. 1

**Comment by hiranya911** [comment]
I think, in the first case you're reading from the `values/a`, so you get the object `{randomValue: "set", sorter: 1}` as the result. In the second case you're reading from `values`, so you get `{a: {randomValue: "set", sorter: 1}}` as the result. You should be able to access the fields as `snapshot.val().a.randomValue`.

This behavior seems correct to me.

**Comment by gNiKnet** [comment]
The problem is that I cannot know that "a" will be the result and it should not always. I am trying to **query**. I found a hack to solve this.

https://stackoverflow.com/questions/45543072/firebase-admin-database-snapshot-data-after-query/45577725#45577725#answer-45577725

I used Object.keys(snapshot.val())[0], which just does not seem like a nice thing to do to me. Anyways thanks for you help, I would love to see a nicer solution for this anyways because I see querying and then doing something with it as a rather bigger usecase.
